src/clj/ataru/applications/application_service.clj

```diff
@@ -104,36 +104,43 @@
 (defn get-application-with-human-readable-koodis
   "Get application that has human-readable koodisto values populated
    onto raw koodi values."
-  [application-key session organization-service tarjonta-service ohjausparametrit-service person-client]
+  [application-key session organization-service tarjonta-service ohjausparametrit-service person-client with-newest-form?]
   (when-let [application (aac/get-latest-application-by-key
                            organization-service
                            tarjonta-service
                            session
                            application-key)]
-    (let [tarjonta-info (tarjonta-parser/parse-tarjonta-info-by-haku
-                         tarjonta-service
-                         organization-service
-                         ohjausparametrit-service
-                         (:haku application)
-                         (:hakukohde application))
-          form-key      (->> (-> tarjonta-info :tarjonta :hakukohteet)
-                             (map :form-key)
-                             (distinct)
-                             (remove nil?)
-                             first)
-          form          (-> (:form application)
-                            form-store/fetch-by-id
-                            koodisto/populate-form-koodisto-fields
-                            (populate-hakukohde-answer-options tarjonta-info)
-                            (hakija-form-service/populate-can-submit-multiple-applications tarjonta-info))
-          newest-form   (some-> form-key form-store/fetch-by-key
-                          (assoc :content []) (dissoc :organization-oid))]
+    (let [tarjonta-info    (tarjonta-parser/parse-tarjonta-info-by-haku
+                            tarjonta-service
+                            organization-service
+                            ohjausparametrit-service
+                            (:haku application)
+                            (:hakukohde application))
+          form-key         (->> (-> tarjonta-info :tarjonta :hakukohteet)
+                                (map :form-key)
+                                (distinct)
+                                (remove nil?)
+                                first)
+          applied-form     #(-> (:form application)
+                               form-store/fetch-by-id)
+          newest-form      (some-> form-key form-store/fetch-by-key)
+          form             (-> (if with-newest-form?
+                                 (or newest-form
+                                     (applied-form))
+                                 (applied-form))
+                               koodisto/populate-form-koodisto-fields
+                               (populate-hakukohde-answer-options tarjonta-info)
+                               (hakija-form-service/populate-can-submit-multiple-applications tarjonta-info))
+          alternative-form (some-> (if (and newest-form with-newest-form?)
```

Comment:
Tässä vois käyttää `if`in sijaista `when`, ja ehkä tehdä myös tuo paluuarvossa suoritettu `:id`-vertailu jo tässä vaiheessa?